src/billing/components/BillingPageContents.tsx

```diff
@@ -10,13 +13,30 @@ import MarketplaceBilling from 'src/billing/components/marketplace/MarketplaceBi
 // Utils
 import {getQuartzMe} from 'src/me/selectors'
 
+// Thunks
+import {getBillingProviderThunk} from 'src/identity/actions/thunks'
+import {shouldUseQuartzIdentity} from 'src/identity/utils/shouldUseQuartzIdentity'
+
 const BillingPageContents: FC = () => {
+  const dispatch = useDispatch()
   const quartzMe = useSelector(getQuartzMe)
 
+  useEffect(() => {
+    if (!CLOUD) {
+      return
+    }
+
+    if (shouldUseQuartzIdentity() && !quartzMe.billingProvider) {
+      dispatch(getBillingProviderThunk())
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
   if (
     (quartzMe?.accountType === 'pay_as_you_go' ||
       quartzMe?.accountType === 'contract') &&
-    quartzMe?.billingProvider !== 'zuora'
+    // This additional check is needed because, on page load, billingProvider (esp. with /identity on) may be 'null', which !== 'zuora. A 'null' billingProvider corresponds to free accounts.
+    quartzMe.billingProvider !== null &&
+    quartzMe.billingProvider !== 'zuora'
```

Comment:
Don't think we need the additional case since `billingProvider` is only `null` [when the account type is free.](https://influxdata.slack.com/archives/CHZ011F7E/p1655303863607759)